### PR TITLE
Shapella submit types

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: pkg/relay/relay.go [hunk 1]
```diff
@@ -209,48 +212,82 @@ func (rs *Relay) GetHeader(ctx context.Context, m *structs.MetricGroup, request
...
-	}
+	fork := rs.beaconState.GetFork(uint64(slot.Epoch()))
 
-	tSignature := time.Now()
-	signature, err := types.SignMessage(&bid, rs.config.BuilderSigningDomain, rs.config.SecretKey)
-	m.AppendSince(tSignature, "getHeader", "signature")
-	if err != nil {
-		return nil, ErrInternal
-	}
+	if fork == structs.ForkBellatrix {
```
Comment (aratz-lasa) [inline pkg/relay/relay.go:217]: Why not switch cases?

File: pkg/relay/relay.go [hunk 2]
```diff
@@ -261,28 +298,32 @@ func (rs *Relay) GetPayload(ctx context.Context, m *structs.MetricGroup, payload
...
 	if err != nil || !ok {
 		return nil, ErrInvalidSignature
 	}
 	m.AppendSince(tVerify, "getPayload", "verify")
 
 	tGet := time.Now()
-	key := structs.PayloadKey{
-		BlockHash: payloadRequest.Message.Body.ExecutionPayloadHeader.BlockHash,
+
+	key := payloadRequest.ToPayloadKey(pk)
```
Comment (aratz-lasa) [inline pkg/relay/relay.go:321]: Love it! <3

File: pkg/relay/relay.go [hunk 3]
```diff
@@ -294,53 +335,60 @@ func (rs *Relay) GetPayload(ctx context.Context, m *structs.MetricGroup, payload
...
 	}
 
 	// defer put delivered datastore write
-	go func(rs *Relay, slot structs.Slot, trace structs.DeliveredTrace) {
+	go func(rs *Relay, slot structs.Slot, payloadRequest structs.SignedBlindedBeaconBlock, trace structs.DeliveredTrace) {
 		if rs.config.PublishBlock {
-			beaconBlock := structs.SignedBlindedBeaconBlockToBeaconBlock(payloadRequest, payload.Payload.Data)
-			if err := rs.beacon.PublishBlock(beaconBlock); err != nil {
-				logger.WithError(err).Warn("fail to publish block to beacon node")
+			beaconBlock, err := payloadRequest.ToBeaconBlock(payload.Payload.Data)
```
Comment (aratz-lasa) [inline pkg/relay/relay.go:359]: <3

**Review by aratz-lasa — APPROVED** [review]
LGTM, but this is a VERY big PR, so we should pull @dmarzzz @lthibault , so that when they have time they review this.

File: pkg/relay/relay.go (hunk 1 above)
Comment (lukanus) [reply pkg/relay/relay.go:217]: Because it's too big to be fair, We can pack this into functions potentially, and do switch but this after that PR

**Pull request merged**